src/App.tsx

```diff
@@ -15,36 +15,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  None,
+  Alphabet,
+  Length,
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const getPreparedGoods = () => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (sortField === SortType.Alphabet) {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (sortField === SortType.Length) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    return sortedGoods;
+  };
+
+  const reset = () => {
+    setSortField(SortType.None);
+    setIsReversed(false);
+  };
+
+  const visibleGoods = getPreparedGoods();
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortField(SortType.Alphabet)}
+          type="button"
+          className={`button is-info ${sortField !== SortType.Alphabet ? 'is-light' : ''}`}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortField(SortType.Length)}
+          type="button"
+          className={`button is-success ${sortField !== SortType.Length ? 'is-light' : ''}`}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setIsReversed(!isReversed)}
+          type="button"
+          className={`button is-warning ${!isReversed ? 'is-light' : ''}`}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField !== SortType.None || isReversed) && (
+          <button
+            onClick={reset}
+            type="button"
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleGoods.map(good => (
+          <li data-cy="Good" key={good}>
```

Comment:
The `key` prop for list items should be unique to each item to ensure React can efficiently update and render components. Currently, the `good` string is used as the key, which may not be unique if there are duplicate goods. Consider using a combination of the item and its index or another unique identifier.